src/pyspiffe/bundle/jwt_bundle/jwt_bundle.py

```diff
@@ -51,7 +65,10 @@ def find_jwt_authority(self, key_id: str) -> _PUBLIC_KEY_TYPES:
         if not key_id:
             raise ValueError('key_id cannot be empty.')
 
-        key = self._jwt_authorities.get(key_id, None)
-        if not key:
-            raise AuthorityNotFoundError(key_id)
-        return key
+        with self.lock:
+            key = self._jwt_authorities.get(key_id)
+
+            if not key:
+                raise AuthorityNotFoundError(key_id)
```

Comment:
this is indeed a good point Max... in either way, I think we need to be consistent in get_jwt_bundle_for_trust_domain and here... (unless, of course, there is a good reason not to). Deserves some research and I don't know the answer to the most pythonic way :).

Reply:
More than if it's pythonic or not pythonic, I think the point here is where we want to handle an empty response.
Not only `get_jwt_bundle_for_trust_domain` handles this scenario differently, `x509Bundle.get_x509_bundle_for_trust_domain()`  also do so.
I'll make this change so we are consistent. Thanks.

Reply:
I think throwing an Exception in case of missing a key_id is something that we brought here from Java, as in the java world returning an Optional is still not generally accepted, but I feel comfortable using it here.

Reply:
Max, agree... the get from dict returns a default value in case key is not present... [key] raises an exception. Seems to me this should be aligned to the get behaviour of Dict. If we proceed down this road, should we also allow for the caller to inform the default to use in case key is not present? I like this capability.

Reply:
yeah, the get from the dictionary returns None in case of a missing key, that's the default value, except that you specify another default value. Not sure if we need to provide a `get_jwt_authority(key, default_authority)`, probably not, it's not present in the other libraries.

```
def test_missing_key():
    mydict = {}
    assert mydict.get('mising key') is None
```